src/ZkFold/Symbolic/Data/ByteString.hs

```diff
@@ -86,15 +331,104 @@ instance (Arithmetic a, KnownNat n) => Arithmetizable a (ByteString n (Arithmeti
 
     typeSize = minNumberOfRegisters @a @n
 
-binOp
-    :: forall a n .
-       Arithmetic a
+
+-- TODO: I really don't like that summation is implemented here and in UInt. Can we do something about it?
```

Comment:
And I don't like that `ByteString` is used for addition as
1. An efficient implementation would require `ByteString` to reserve space just as in `UInt` (and why not use `UInt` then)
2. The current implementation which uses a bitwise summation is infeasible